### PR TITLE
Issue 8 firebase database setup

### PR DESCRIPTION
**Pull request by fluffyemily**
Add Firebase 3
Add Geofire (couldn't use cocoapods, Firebase SDK version and Geofire cocoapod were incompatible, so have included Geofire directly as thirdparty code)
Setup and authorise Firebase when app starts

#8 

Based of a #7

### CODE REVIEW

**Comment by fluffyemily** [comment]
[Integrate Firebase](https://trello.com/c/yGaE9t9J/97-integrate-firebase)

**Comment by mcomella** [comment]
> Firebase SDK version and Geofire cocoapod were incompatible, so have included Geofire directly as thirdparty code

I'm not sure I understand – so the Firebase SDK pod and the Geofire pod in Cocoapods are incompatible? That's concerning. I did a brief search and it seems people were having maybe related problems relatively recently ([link](https://github.com/firebase/geofire-objc/issues/48)).

File: Prox/Prox/AppDelegate.swift [hunk 1]
```diff
@@ -23,9 +26,28 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
         return true
     }
 
+    private func setupFirebase() {
+        FIRApp.configure()
+
+        if let user = FIRAuth.auth()?.currentUser {
+            authorizedUser = user
+        } else {
+            FIRAuth.auth()?.signInAnonymously { (user, error) in
```
Comment (mcomella) [inline Prox/Prox/AppDelegate.swift]: Interesting – I only see [`signInAnonymouslyWithCompletion` in the docs](https://firebase.google.com/docs/reference/ios/firebaseauth/interface_f_i_r_auth.html#a0ffa43acd046e5ee42967106f73556f6) – am I looking at the right version of the docs?

**Comment by mcomella** [comment]
Note the Buddybuild failure though!

File: Prox/Prox/AppDelegate.swift (hunk 1 above)
Comment (fluffyemily) [inline Prox/Prox/AppDelegate.swift]: yes you are. It's an interesting feature of swift, if the final argument of a function is a closure, you can define that closure at the end of the function, not inside it, and you don't have to specify it's var name. This _is_  [`signInAnonymouslyWithCompletion`](https://firebase.google.com/docs/reference/ios/firebaseauth/interface_f_i_r_auth.html#a0ffa43acd046e5ee42967106f73556f6)

**Pull request merged**